From f02eb2b8687f78a644369a98290db0501bdff7f4 Mon Sep 17 00:00:00 2001
From: Vishal Sagar <vishal.sagar@xilinx.com>
Date: Fri, 6 Mar 2020 22:12:27 -0800
Subject: [PATCH 1265/1852] v4l: xilinx: sdirxss: Fix returning error on stream
 off

commit 18fa67a01e060e080cfd3b1dc9821ca9e38ab917 from
https://github.com/Xilinx/linux-xlnx.git

When yavta app is started and the cable removed, a video unlock event
occurs. In this context the streaming is stopped as the bridges are
disabled. Now killing the yavta application calls s_stream() with enable
as false which returns -EINVAL.
So when yavta is restarted after cable is connected and video is locked,
the V4L framework doesn't call s_stream() with enable = true as it still
believes that SDI Rx subdev is still streaming.

This is fixed by returning success for s_stream() with enable = false
even when streaming is disabled.

Signed-off-by: Vishal Sagar <vishal.sagar@xilinx.com>
Reviewed-by: Hyun Kwon <hyun.kwon@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/media/platform/xilinx/xilinx-sdirxss.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/media/platform/xilinx/xilinx-sdirxss.c b/drivers/media/platform/xilinx/xilinx-sdirxss.c
index e60a9da75c5c..3e881dc80198 100644
--- a/drivers/media/platform/xilinx/xilinx-sdirxss.c
+++ b/drivers/media/platform/xilinx/xilinx-sdirxss.c
@@ -1552,7 +1552,7 @@ static int xsdirxss_s_stream(struct v4l2_subdev *sd, int enable)
 	} else {
 		if (!xsdirxss->streaming) {
 			dev_dbg(core->dev, "Stopped streaming already\n");
-			return -EINVAL;
+			return 0;
 		}
 
 		xsdirx_streamflow_control(core, false);
-- 
2.31.1

